Sources/Harbor/HRequestManager.swift

```diff
@@ -20,23 +51,13 @@ extension HRequestManager {
             return .error(.noConnectionError)
         }
 
-        if request.needsAuth {
-            if let authCredential = await Self.config.authProvider?.getAuthorizationHeader() {
-                var mutableRequest = request
-                var headerParameters = request.headerParameters ?? [:]
-                headerParameters[authCredential.key] = authCredential.value
-                mutableRequest.headerParameters = headerParameters
-
-                let requestCopy = mutableRequest
-                async let result = self.requestHandler(model: model, request: requestCopy)
-                return await result
-            } else {
-                return .error(.authProviderNeeded)
-            }
-        } else {
-            async let result = self.requestHandler(model: model, request: request)
-            return await result
+        var modifiedRequest = request
+        if request.needsAuth, let authCredential = await Self.config.authProvider?.getAuthorizationHeader() {
+            modifiedRequest.headerParameters?[authCredential.key] = authCredential.value
```

Comment:
If headerParameters is nil, you are not setting the header. For that reason I was doing this:
`var headerParameters = request.headerParameters ?? [:]`